File: LLM-Translations/GPT-4/avatar/Java/Go/atcoder_AGC004_B.go
package main

import (
	"fmt"
	"math"
	"bufio"
	"os"
	"strconv"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	input := strings.Split(scanner.Text(), " ")
	N, _ := strconv.Atoi(input[0])
	x, _ := strconv.ParseInt(input[1], 10, 64)

	srimeList := make([]int64, N)
	scanner.Scan()
	input = strings.Split(scanner.Text(), " ")
	for i := 0; i < N; i++ {
		srimeList[i], _ = strconv.ParseInt(input[i], 10, 64)
	}

	costList := make([][]int64, N)
	for i := 0; i < N; i++ {
		costList[i] = make([]int64, N)
	}

	for i := 0; i < N; i++ {
		costList[i][0] = srimeList[i]
		for castNum := 1; castNum < N; castNum++ {
			costList[i][castNum] = int64(math.Min(float64(costList[i][castNum-1]), float64(srimeList[(i-castNum+N)%N])))
		}
	}

	result := make([]int64, N)
	for i := 0; i < N; i++ {
		sum := int64(0)
		for _, costArray := range costList {
			sum += costArray[i]
		}
		result[i] = int64(i) * x + sum
	}

	min := result[0]
	for _, val := range result {
		if val < min {
			min = val
		}
	}

	fmt.Println(min)
}

// //End of Code.

File: LLM-Translations/GPT-4/avatar/Java/C/atcoder_AGC004_B.c
#include <stdio.h>
#include <stdlib.h>
#include <limits.h>

int main() {
    int N;
    long x;
    scanf("%d %ld", &N, &x);
    long srimeList[N];
    for (int i = 0; i < N; i++) {
        scanf("%ld", &srimeList[i]);
    }
    long costList[N][N];
    for (int i = 0; i < N; i++) {
        costList[i][0] = srimeList[i];
        for (int castNum = 1; castNum < N; castNum++) {
            costList[i][castNum] = (costList[i][castNum - 1] < srimeList[(i - castNum + N) % N]) ? costList[i][castNum - 1] : srimeList[(i - castNum + N) % N];
        }
    }
    long result[N];
    for (int i = 0; i < N; i++) {
        long sum = 0;
        for (int j = 0; j < N; j++) {
            sum += costList[j][i];
        }
        result[i] = i * x + sum;
    }
    long min = LONG_MAX;
    for (int i = 0; i < N; i++) {
        if (result[i] < min) {
            min = result[i];
        }
    }
    printf("%ld\n", min);
    return 0;
} // //End of Code
